File: app/src/main/java/com/android2ee/training/freenet/view/SimpleActivity.java
package com.android2ee.training.freenet.view;
// 

import android.app.Activity;
import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.android2ee.training.freenet.R;
import com.android2ee.training.freenet.repository.AndroidDatabase;

import java.io.File;
import java.io.FileInputStream;

/**
 * Created by Mathias Seguy also known as Android2ee on 27/03/2020.
 * The goal of this class is to :
 */
public class SimpleActivity extends Activity {

    AndroidDatabase database;
    ImageView imageView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_simple);

        this.database = new AndroidDatabase(this);

        ((TextView) findViewById(R.id.simple_text)).setText(getName(this));
        imageView = (ImageView) findViewById(R.id.simple_image);

        final TextView asynchronousTextView = (TextView) findViewById(R.id.simple_async_text);
        //Since Froyo, we said you should not use asynctask
        //not only for the memory leaks
        new AsyncTask<Void, Void, String>() {
            @Override
            protected String doInBackground(Void... voids) {
                return database.getDescription();
            }

            @Override
            protected void onPostExecute(String result) {
                asynchronousTextView.setText(result);
            }
        }.execute();
    }

    @Override
    protected void onResume() {
        super.onResume();
        imageView.setImageBitmap(getPicture());
    }

    public String getName(Context context) {
        String contactFormat = context.getString(R.string.contact_format); // Contact: %s
        return String.format(contactFormat, getIntent().getStringExtra("name"));
    }

    public Bitmap getPicture() {
        File image = new File(this.getCacheDir(), getIntent().getStringExtra("name") + ".jpg");
        Bitmap bitmap = null;
        try {
            byte[] buf = new byte[(int) image.length()];
            FileInputStream inputStream = new FileInputStream(image);
            inputStream.read(buf);
            bitmap = BitmapFactory.decodeByteArray(buf, 0, (int) image.length());
        } catch (Exception e) {
            Log.e("picture", e.getMessage());
        }

        return bitmap;
    }
}

File: app/src/main/java/com/android2ee/training/freenet/view/simple/SimpleActivity.kt
package com.android2ee.training.freenet.view.simple

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.android2ee.training.freenet.R

//
/**
 * Created by Mathias Seguy also known as Android2ee on 27/03/2020.
 * The goal of this class is to :
 */
class SimpleActivity : AppCompatActivity() {
    /***********************************************************
     *  Attributes
     **********************************************************/
    /**     * Model       */
    lateinit var model: SimpleActivityViewModel

    /** Lazy load you widget */
    private val imvPicture: AppCompatImageView by lazy {
        findViewById<AppCompatImageView>(R.id.simple_image)
    }
    private val txvSimpleText: AppCompatTextView by lazy {
        findViewById<AppCompatTextView>(R.id.simple_text)
    }
    private val txvSimpleTextAsync: AppCompatTextView by lazy {
        findViewById<AppCompatTextView>(R.id.simple_async_text)
    }

    /** The name */
    val name: String by lazy {
        intent.getStringExtra("name")
    }

    /***********************************************************
     *  Managing LifeCycle
     **********************************************************/
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //load the layout
        setContentView(R.layout.activity_simple)
        //load your model
        model = ViewModelProvider(this).get(SimpleActivityViewModel::class.java)
    }

    override fun onStart() {
        super.onStart()
        //register as observer of the data you are interested in
        model.mDescriptionState.observe(this, Observer { desc -> txvSimpleTextAsync.text = desc })
        model.mPictureState.observe(this, Observer { bitmap ->
            if (bitmap != null) {
                imvPicture.setImageBitmap(bitmap)
            } else {
                imvPicture.setImageResource(R.drawable.ic_rowing)
            }
        })
    }

    override fun onResume() {
        super.onResume()
        //Load your picture
        model.getPictureAsync(name)
        txvSimpleText.text = name
    }
}